packages/node/examples/fastify/index.js

```diff
@@ -21,16 +34,16 @@ fastify.addHook('onSend', async (request, reply, payload) => {
 
     startedDateTime: new Date(request.readmeStartTime),
     responseEndDateTime: new Date(),
-    responseBody: payload,
+    responseBody: reply.payload,
```

Comment:
What's the difference between the `payload` in `onResponse` hooks versus the one in `onSend`?

Reply:
Yeah great question, and it took me a good couple of hours to figure out the right hook to use here. onSend is called right before any response has been returned, so the data is there but there's no content-length header set up yet, so the har file has a `content.size` of 0 https://github.com/readmeio/metrics-sdks/blob/c093e21af1fc9b659fea62fdc5ac93fc9dabfd99/packages/node/src/lib/process-response.ts#L68. 

onResponse is called after the response has been sent, but you can't access the body at that point... hence why we've gotta attach the payload in the onSend, then process it after the response has gone down.

The reason why we have to pass `request.raw` to the log function is because fastify's Request wrapper doesn't have `request.httpVersion` so [this](https://github.com/readmeio/metrics-sdks/blob/c093e21af1fc9b659fea62fdc5ac93fc9dabfd99/packages/node/src/lib/process-request.ts#L195) was coming through as `HTTP/undefined`.

The reason why we have to pass the [Reply](https://www.fastify.io/docs/latest/Reference/Reply/) instance to the log function (as opposed to `reply.raw`) is also nuanced, but basically Node's ServerResponse object removes all headers from the object after it's been written to the response, so `res.getHeader('content-length')` would return with undefined when the response has finished.

I hope this makes sense 😅 it didn't make a ton of sense to me tbh, hence why i'm hoping to modify the API design to make all of this stuff a little easier to figure out.